replica/replica.go

```diff
@@ -213,6 +218,29 @@ func (r *Replica) diskPath(name string) string {
 	return path.Join(r.dir, name)
 }
 
+func (r *Replica) isExtentSupported() error {
+	filePath := r.diskPath(r.info.Head + metadataSuffix)
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return err
+	}
+
+	file, err := os.OpenFile(filePath, os.O_RDONLY, 0666)
+	if err != nil {
+		if os.IsPermission(err) {
+			logrus.Errorf("failed to read %s, error: Read permission denied", filePath)
+		}
+		return err
+	}
+
+	defer file.Close()
+
+	fiemapFile := fibmap.NewFibmapFile(file)
+	if _, err := fiemapFile.Fiemap(uint32(fileInfo.Size())); err != 0 {
```

Comment:
`err != 0` implies that err is a number. Then what's `err.Error()`?

Also, if `err` is indeed a number, use `errno` instead of `err`.

Reply:
thanks for pointing this out.

Reply:
it returns `operation not supported` and `0` if there is no error.

Reply:
Dig around for a bit. Seems Fiemap should box `errno` to `error` before returning, using something similar to `errnoErr` in https://golang.org/src/syscall/syscall_unix.go. I guess we need to live with `errno != 0` for now.